docs/tutorials/01_neural_networks.ipynb

```diff
@@ -489,7 +767,6 @@
   },
   {
    "cell_type": "markdown",
-   "id": "according-watch",
    "metadata": {},
    "source": [
     "## 4. `CircuitQNN`\n",
```

Comment:
I think that the circuit QNN intro was originally convoluted, and I would replace it with something like: 

```
The `CircuitQNN` is defined using a single (parametrized) `QuantumCircuit`. This circuit represents both the feature map and ansatz of the QNN. At first sight, it might look similar to the previously defined `TwoLayerQNN`, but there is one fundamental difference. `TwoLayerQNN`s return the evaluated expectation values with respect to a custom operator. The output format is fixed by this custom operator, which defaults to parity (i.e a single value between 0 and 1 per input, also known as binary output). 

On the other hand, `CircuitQNN`s are evaluated by performing measurements (getting samples) on the underlying quantum circuit, and offer a wider variety of output formats that can be configured during their initialization. 

The samples can either be interpreted as probability distributions or directly as a batch of binary output. In the case of probabilities, gradients can be estimated efficiently and the CircuitQNN provides a backward pass as well. In case of samples, differentiation is not possible and the backward pass returns (None, None).

Furthermore, the CircuitQNN allows to specify an interpret function to post-process the samples. This is expected to take a measured integer (from a bitstring) and map it to a new index, i.e. non-negative integer. In this case, the output shape needs to be provided and the probabilities are aggregated accordingly.

A CircuitQNN can be configured to return sparse as well as dense probability vectors. As we will see soon, if no interpret function is used, the dimension of the probability vector scales exponentially with the number of qubits, and a sparse probability vector is usually recommended. 
```

Reply:
Done